internal/boxcli/shellenv.go

```diff
@@ -85,7 +89,10 @@ func shellEnvFunc(cmd *cobra.Command, flags shellEnvCmdFlags) (string, error) {
 		}
 	}
 
-	envStr, err := box.NixEnv(cmd.Context(), flags.runInitHook)
+	envStr, err := box.NixEnv(cmd.Context(), devopt.NixEnvOpts{
+		DontRecomputeEnvironment: !recomputeEnvIfNeeded,
```

Comment:
nbd, but I prefer the `recomputeEnvIfNeeded` as a param name. Can we reuse that within `NixEnvOpts` as well?

Alternatively, could it be `skipRecomputeEnvironment`?

Reply:
Yeah I struggled with this as well. Here's why I went with this:

The issue with `recomputeEnvIfNeeded` is that we would need to default it to `true`. I prefer if the default is true and only pass in a param to make it false.

The reason that flag is the other way around is because `devbox global shellenv --no-recompute=false` would be super awkward because of the double negative.